egs2/chime8_task1/asr1/run.sh

```diff
@@ -0,0 +1,284 @@
+          --nj $ngpu \
+          --max-segment-length $max_segment_length \
+          --max-batch-duration $gss_max_batch_dur \
+          --channels $channels \
+          --use-selection $use_selection \
+          --top-k $top_k \
+          --gss-iterations $gss_iterations
+    log "Guided Source Separation processing for ${dset_name}/${dset_part} was successful !"
+  done
+fi
+
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
+  # Decoding on dev set because test is blind for now
+
+  pretrained_affix=
+  if [ -n "$use_pretrained" ]; then
+    pretrained_affix+="--skip_data_prep false --skip_train true "
+    pretrained_affix+="--download_model ${use_pretrained}"
+  fi
+
+
+  if [ -z $diar_score ] || [ $decode_train != "train" ]; then
+    asr_dprep_stage=3
+  fi
+
+  # these are args to ASR data prep, done in local/data.sh
+  data_opts="--stage $asr_dprep_stage --chime6-root ${chime6_root} --train-set ${asr_train_set}"
+  data_opts+=" --manifests-root $manifests_root --gss_dsets $gss_dsets --gss-dump-root $gss_dump_root"
+  data_opts+=" --decode-train ${decode_train}"
+  # override ASR conf/tuning to scale automatically with num of GPUs
+  asr_args="--batch_size ${asr_batch_size} --scheduler_conf warmup_steps=${asr_warmup}"
+  asr_args+=" --max_epoch=${asr_max_epochs} --optim_conf lr=${asr_max_lr}"
+
+  ./asr.sh \
+    --lang en \
+    --local_data_opts "${data_opts}" \
+    --stage $asr_stage \
+    --ngpu $ngpu \
+    --asr_args "${asr_args[@]}" \
+    --token_type bpe \
+    --nbpe $nbpe \
+    --bpe_nlsyms "${bpe_nlsyms}" \
+    --nlsyms_txt "data/nlsyms.txt" \
+    --feats_type raw \
+    --feats_normalize utterance_mvn \
+    --audio_format "flac" \
+    --min_wav_duration $train_min_segment_length \
+    --max_wav_duration $train_max_segment_length \
+    --speed_perturb_factors "0.9 1.0 1.1" \
+    --asr_config "${asr_config}" \
+    --inference_config "${inference_config}" \
+    --use_lm ${use_lm} \
+    --lm_config "${lm_config}" \
+    --inference_asr_model ${inference_asr_model} \
+    --use_word_lm ${use_word_lm} \
+    --word_vocab_size ${word_vocab_size} \
+    --train_set "${asr_train_set}" \
+    --valid_set "${asr_cv_set}" \
+    --test_sets "${asr_tt_set}" \
+    --bpe_train_text "data/${asr_train_set}/text" \
+    --lm_train_text "data/${asr_train_set}/text" ${pretrained_affix}
+fi
+
+if [ "${decode_train}" == "eval" ]; then
+  log "Scoring not available for eval set till the end of the challenge."
+  exit
+fi
+
+if [ ${stage} -le 4 ] && [ ${stop_stage} -ge 4 ]; then
+  # final scoring
+  log "Scoring ASR predictions for CHiME-7 DASR challenge."
+  # note, we re-create the asr exp folder here based on asr.sh
+  if [ -n "$use_pretrained" ]; then
+    asr_exp="exp/${use_pretrained}"
+  else
+    #FIXME ask kamo ? if this fails put your own !
+    asr_tag="$(basename "${asr_config}" .yaml)_raw"
+    asr_exp="exp/asr_${asr_tag}"
+  fi
+  inference_tag="$(basename "${inference_config}" .yaml)"
```

Comment:
This is currently failing BTW. 
basically due to bash overriding the learning rate and batch size the name of the trained model will be different than the name of the YAML file. 
Is there a way to set the name for the ASR model folder ? @simpleoier

Reply:
As we discussed, this is due to the difference of asr_tag here in this file and asr.sh.